src/routing/types.ts

```diff
@@ -0,0 +1,13 @@
+export interface RouterProps {
+  children: Array<PageType>;
+  basename: string;
+}
+
+export interface PageType extends React.Component {
+  props: {
+    path: string; // Used by the Router directly
+    exact?: boolean;
```

Comment:
I might recommend splitting this off into a `Routable` type and perform a type intersection (`&`).
```ts
props: Routable & {
  children: React.Component;
  title: string;
}
```